research/flamby/fed_isic2019/mr_mtl_mkmmd/client.py

```diff
@@ -0,0 +1,199 @@
+import argparse
+import os
+from logging import INFO
+from pathlib import Path
+from typing import Optional, Sequence, Tuple
+
+import flwr as fl
+import torch
+import torch.nn as nn
+from flamby.datasets.fed_isic2019 import BATCH_SIZE, LR, NUM_CLIENTS, Baseline, BaselineLoss
+from flwr.common.logger import log
+from flwr.common.typing import Config
+from torch.nn.modules.loss import _Loss
+from torch.optim import Optimizer
+from torch.utils.data import DataLoader
+
+from fl4health.checkpointing.checkpointer import BestLossTorchCheckpointer
+from fl4health.checkpointing.client_module import ClientCheckpointModule
+from fl4health.clients.mkmmd_clients.mr_mtl_mkmmd_client import MrMtlMkMmdClient
+from fl4health.utils.losses import LossMeterType
+from fl4health.utils.metrics import BalancedAccuracy, Metric
+from fl4health.utils.random import set_all_random_seeds
+from research.flamby.flamby_data_utils import construct_fedisic_train_val_datasets
+
+FED_ISIC2019_BASELINE_LAYERS = []
+for i in range(16):
+    FED_ISIC2019_BASELINE_LAYERS.append(f"base_model._blocks.{i}")
+FED_ISIC2019_BASELINE_LAYERS += ["base_model._dropout"]
```

Comment:
This is more of a philosophical question, but do you think we should consider features before or after dropout in these settings? Part of me feels like before, since drop out will add stochasticity to the feature maps but I'm not sure.

Reply:
Yeah good question. I feel like I also agree that pre dropout is better. So I will drop it.